core/src/main/java/com/linecorp/armeria/client/Clients.java

```diff
@@ -477,7 +477,7 @@ public static SafeCloseable withHttpHeaders(
         requireNonNull(headerManipulator, "headerManipulator");
         return withContextCustomizer(ctx -> {
             final HttpHeaders manipulatedHeaders = headerManipulator.apply(ctx.additionalRequestHeaders());
-            ctx.setAdditionalRequestHeaders(manipulatedHeaders);
+            ctx.mutateAdditionalRequestHeaders(mutator -> mutator.add(manipulatedHeaders));
```

Comment:
In this class, we have 3 different `withHttpHeader` methods and I'm wondering if we can do the same approach here. Or at least we can deprecate this method and add a new method that takes a `Consumer`.

Reply:
It's pretty common to specify a single header, so I'd prefer keeping them all.

Reply:
Then just how about changing this method?
```java
// before
public static SafeCloseable withHttpHeaders(
            Function<? super HttpHeaders, ? extends HttpHeaders> headerManipulator) {
// after
public static SafeCloseable withHttpHeaders(Consumer<HttpHeadersBuilder> headerManipulator) {
```
So that a user does not have to `toBuilder` and `build`?

Reply:
OK!

Reply:
Done! 😉